activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -109,7 +112,7 @@ def default_insert_value(column)
           end
 
           def last_inserted_id(result)
-            @raw_connection.last_id
+            @raw_connection&.last_id
```

Comment:
I think this is technically a tiny bleed-over of #44591 too -- this change alone doesn't introduce any occasions where `@raw_connection` can be nil. This is a notable occasion where we do want to still use the raw ivar, though, and _not_ [the upcoming definition of] `any_raw_connection`: if we don't currently have a connection, there's no point establishing one just to ask what it last inserted, because the answer will definitely be nil.

This is not a behaviour change: a connection that hasn't yet performed an insert will have a nil `last_id`.